### DISCUSSION TITLE
WFDS

### DISCUSSION THREAD

**Discussion by Weggins** [firemodels/fds#11768] [General]
Hi everyone,
   I am trying to establish a model to simulate the fire spread on a canyon terrain. I use 'GEOM' to realize a canyon terrain and I use the parameters about vegetation which is referred to in hill_structure_bench.fds. However, it is a pity that I cannot see the fire spread. I hope someone can give me some advise. The commands are the following:
&MESH ID='canyon', IJK=80,70,50, XB=-4.0,4.0,-1.0,6.0,0.0,5.0/

&REAC ID='WOOD'
      FUEL='WOOD'
      FYI='Ritchie, et al., 5th IAFSS, C_3.4 H_6.2 O_2.5, dHc = 15MW/kg'
      SOOT_YIELD = 0.02
      O          = 2.5
      C          = 3.4
      H          = 6.2
      HEAT_OF_COMBUSTION = 17700 /

&SPEC ID='WATER VAPOR' /

- Australian grass based on experimental case F19 (Mell et al., 2007 IJWF)

&SURF ID = 'GRASS'
      MATL_ID = 'Dry Grass'
      MASS_PER_VOLUME = 12
      THICKNESS= 0.51
      MOISTURE_FRACTION= 0.06
      SURFACE_VOLUME_RATIO = 12240
      EMISSIVITY = 0.99
      RGB = 122,117,48
  /

&MATL ID                 = 'AIR'
      DENSITY            = 1.2
      CONDUCTIVITY       = 0.026
      SPECIFIC_HEAT      = 1.01 /

&MATL ID                 = 'MOISTURE'
      DENSITY            = 997.
      CONDUCTIVITY       = 0.62
      SPECIFIC_HEAT      = 4.184
      A                  = 600000.
      E                  = 48200.
      N_T                = -0.5
      SPEC_ID            = 'WATER VAPOR'
      NU_SPEC            = 1.0
      HEAT_OF_REACTION   = 2259. /

&MATL ID                 = 'Dry Grass'
      DENSITY            = 512.
      CONDUCTIVITY       = 0.11
      SPECIFIC_HEAT      = 1.5
      A                  = 36300.
      E                  = 60300.
      MATL_ID            = 'Char'
      NU_MATL            = 0.2
      SPEC_ID            = 'WOOD'
      NU_SPEC            = 0.8
      HEAT_OF_REACTION   = 711. /

&MATL ID                 = 'Char'
      DENSITY            = 300.
      CONDUCTIVITY       = 0.052
      SPECIFIC_HEAT      = 1.        /
      N_S                = 0.
      NU_O2_CHAR         = 1.65
      BETA_CHAR          = 0.2
      A                  = 430.
      E                  = 74800.
      MATL_ID            = 'Ash'
      NU_MATL            = 0.0227273
      SPEC_ID            = 'PRODUCTS'
      NU_SPEC            = 0.9965
      HEAT_OF_REACTION   = -32370. /

&MATL ID            = 'Ash'
      DENSITY       = 67.
      CONDUCTIVITY  = 0.1
      SPECIFIC_HEAT = 2.0 /

&GEOM ID='terrain', SURF_ID='GRASS', IS_TERRAIN=T,BNDF_GEOM=T, 
   VERTS=
-3.5,0.0,2.0,
-3.5,6.0,2.0,
-0.1,0.0,0.0,
-0.1,6.0,0.0,
0.1,0.0,0.0,
0.1,6.0,0.0,
3.5,0.0,2.0,
3.5,6.0,2.0,
FACES= 
           1,4,2,1,
           1,3,4,1,
           3,6,4,1, 
           3,5,6,1,
           5,8,6,1,
           5,7,8,1/
&SURF ID='fire',HRRPUA=1000,RAMP_Q='RAMPIGN',RGB=255,0,0,/
&RAMP ID='RAMPIGN',T= 0,F=1 /
&RAMP ID='RAMPIGN',T= 20,F=1 /
&RAMP ID='RAMPIGN',T= 21,F=0 /
&VENT XB=-0.1,0.1,0.5,0.7,0.0,0.0,SURF_ID='fire'/

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /

&BNDF QUANTITY='WALL TEMPERATURE'/
&BNDF QUANTITY='WALL THICKNESS'/
&BNDF QUANTITY='BURNING RATE'/
&BNDF QUANTITY='RADIATIVE HEAT FLUX'/
&BNDF QUANTITY='CONVECTIVE HEAT FLUX'/

&TAIL /

**Comment by ericvmueller** [comment]
Add `GEOM = T` to ensure your ignition is applied to the complex geometry:
`&VENT XB=-0.1,0.1,0.5,0.7,0.0,0.0,SURF_ID='fire', GEOM = T/`

I'd also suggest having it ramp up over the first second or so, rather than start at full HRRPUA. Testing now but it looks like the fire is spreading.

**Reply by ericvmueller** [reply]
Converted to a discussion unless we identify a bug

**Comment by rmcdermo** [comment]
All: Please don't move Issues to Discussions.  I did this too and it causes problems.  Going the other way is fine.  But now actually I would like this to be a feature request where we make this ability to use VENT more automatic for the user.  I don't think a user should need to add GEOM=T here.  I understand it was convenient for get things rolling.  But it is pretty obvious in this case what the user is trying to do.

**Comment by marcosvanella** [comment]
I say we make it a Feature request. We are in the onset of dealing with vents for GEOM.
@Weggins, other than Erics suggestion, try setting one of your triangles SURF indexes to point to 'fire', i.e:

&GEOM ID='terrain', SURF_ID='GRASS',**'fire',** IS_TERRAIN=T,BNDF_GEOM=T,
VERTS= ...
FACES=
1,4,2,**2**,
1,3,4,1,
3,6,4,1,
3,5,6,1, ..

Here 'fire' has been assigned to the first triangle. If you need a smaller ignition region refine your triangulation there.

**Comment by rmcdermo** [comment]
Yes, I forgot, we are working on tying VENT to the GEOM line for HVAC purposes too.  So, this should be connected.

**Comment by ericvmueller** [comment]
Okay - I was looking to see if I could just 'unlock' the previous issue (#11766 ), but I don't see an obvious way. I will open a new feature request and reference this

**Comment by Weggins** [comment]
Hi everyone,
Thanks for your help!
Now the fire can spread successfully. However, I have another problem: the fire just spread upwards. I set the fire source in the middle of the centerline of the canyon, but the fire does not spread downward. Can someone answer me? The picture is following:
![fireline](https://user-images.githubusercontent.com/131326636/234205029-1236ae7e-ed03-4c18-a7d8-5bbb173f13f5.png)

**Comment by Weggins** [comment]
The fire should spread in all directions in experiments and the fire line looks like an elliptic.

**Reply by ericvmueller** [reply]
I find the fire spreads in all directions as you expect. However, the spread is very fast. You have a high mass per volume, and may also need to consider the `MINIMUM_BURNOUT_TIME` parameter used in the grassfire examples (https://github.com/firemodels/fds/blob/master/Validation/CSIRO_Grassland_Fires/FDS_Input_Files/Case_F19_BFM.fds) 

![image](https://user-images.githubusercontent.com/13366803/234309329-92e7bf44-1018-409d-994f-81be831bdd33.png)

**Reply by Weggins** [reply]
Thank you, sir!
I tried to run it again after I check the examples that you give me and revised the parameters of the vegetation, but the fire just spread at the beginning.  Besides, I  do not know how to use 'PART ID' to describe the vegetation in the terrain, I guess that is the reason why the fire cannot spread.
![fire2](https://user-images.githubusercontent.com/131326636/235015490-7e1b25b5-10c2-43e2-9a6a-460f22d97b79.png)
![fire1](https://user-images.githubusercontent.com/131326636/235015498-6030303b-e69d-46c0-9fb0-dce660aaaac2.png)

**Reply by Weggins** [reply]
I notice that the fire also cannot spread in the case of 'hill_structure', which uses the term 'SURF' to describe the vegetation.
&GEOM ID='terrain', SURF_ID='GRASS', IS_TERRAIN=T,
      VERTS=
        20.0,-25.0,0.0,
        20.0, 25.0,0.0,
        24.0,-25.0,4.0,
        24.0, 25.0,4.0,
        28.0,-25.0,4.0,
        28.0, 25.0,4.0,
        31.0,-25.0,0.0,
        31.0, 25.0,0.0,
         0.0,-25.0,0.0,
         0.0, 25.0,0.0,
        50.0,-25.0,0.0,
        50.0, 25.0,0.0,

**Reply by ericvmueller** [reply]
When using all the materials properties for the current grassland cases (https://github.com/firemodels/fds/blob/master/Validation/CSIRO_Grassland_Fires/FDS_Input_Files/Case_F19_BFM.fds), the fire spreads ...though it does begin to die on the lee slope.

@gforney should we make these smv examples consistent with the current validation cases and/or fold into the fds repo?

![image](https://user-images.githubusercontent.com/13366803/235157728-ada72c00-1876-415b-8031-082310d443c1.png)

**Reply by marcosvanella** [reply]
@Weggins I cannot reproduce what you are seeing in your V shaped canyon in any Operating System (OSX, Linux, Windows 10) with 6.8. Please look at my post below and run the case as I posted it to see if you get burning going.

**Comment by marcosvanella** [comment]
@weggins make sure you are using the latest release 6.8.0 https://github.com/firemodels/fds/releases

**Reply by Weggins** [reply]
Thank you, sir!
I just updated FDS to the latest version.

**Comment by marcosvanella** [comment]
@Weggins What OS are you using? 
This is what I see with 6.8 in my Mac computer with the following input file:

&HEAD CHID='canyon' /

&MESH ID='canyon', IJK=80,70,50, XB=-4.0,4.0,-1.0,6.0,0.0,5.0/

&TIME T_END=10/

&REAC ID='WOOD'
FUEL='WOOD'
FYI='Ritchie, et al., 5th IAFSS, C_3.4 H_6.2 O_2.5, dHc = 15MW/kg'
SOOT_YIELD = 0.02
O = 2.5
C = 3.4
H = 6.2
HEAT_OF_COMBUSTION = 17700 /

&SPEC ID='WATER VAPOR' /

! Australian grass based on experimental case F19 (Mell et al., 2007 IJWF)

&SURF ID = 'GRASS'
MATL_ID = 'Dry Grass'
MASS_PER_VOLUME = 12
THICKNESS= 0.51
MOISTURE_FRACTION= 0.06
SURFACE_VOLUME_RATIO = 12240
EMISSIVITY = 0.99
RGB = 122,117,48
/

&MATL ID = 'AIR'
DENSITY = 1.2
CONDUCTIVITY = 0.026
SPECIFIC_HEAT = 1.01 /

&MATL ID = 'MOISTURE'
DENSITY = 997.
CONDUCTIVITY = 0.62
SPECIFIC_HEAT = 4.184
A = 600000.
E = 48200.
N_T = -0.5
SPEC_ID = 'WATER VAPOR'
NU_SPEC = 1.0
HEAT_OF_REACTION = 2259. /

&MATL ID = 'Dry Grass'
DENSITY = 512.
CONDUCTIVITY = 0.11
SPECIFIC_HEAT = 1.5
A = 36300.
E = 60300.
MATL_ID = 'Char'
NU_MATL = 0.2
SPEC_ID = 'WOOD'
NU_SPEC = 0.8
HEAT_OF_REACTION = 711. /

&MATL ID = 'Char'
DENSITY = 300.
CONDUCTIVITY = 0.052
SPECIFIC_HEAT = 1. /
N_S = 0.
NU_O2_CHAR = 1.65
BETA_CHAR = 0.2
A = 430.
E = 74800.
MATL_ID = 'Ash'
NU_MATL = 0.0227273
SPEC_ID = 'PRODUCTS'
NU_SPEC = 0.9965
HEAT_OF_REACTION = -32370. /

&MATL ID = 'Ash'
DENSITY = 67.
CONDUCTIVITY = 0.1
SPECIFIC_HEAT = 2.0 /

&GEOM ID='terrain', SURF_ID='GRASS', IS_TERRAIN=T,BNDF_GEOM=T,
VERTS=
-3.5,0.0,2.0,
-3.5,6.0,2.0,
-0.1,0.0,0.0,
-0.1,6.0,0.0,
0.1,0.0,0.0,
0.1,6.0,0.0,
3.5,0.0,2.0,
3.5,6.0,2.0,
FACES=
1,4,2,1,
1,3,4,1,
3,6,4,1,
3,5,6,1,
5,8,6,1,
5,7,8,1/
&SURF ID='fire',HRRPUA=1000,RAMP_Q='RAMPIGN',RGB=255,0,0,/
&RAMP ID='RAMPIGN',T= 0,F=1 /
&RAMP ID='RAMPIGN',T= 20,F=1 /
&RAMP ID='RAMPIGN',T= 21,F=0 /
&VENT XB=-0.1,0.1,0.5,0.7,0.0,0.0,SURF_ID='fire',GEOM=T/

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /

&BNDF QUANTITY='WALL TEMPERATURE'/
&BNDF QUANTITY='WALL THICKNESS'/
&BNDF QUANTITY='BURNING RATE'/
&BNDF QUANTITY='RADIATIVE HEAT FLUX'/
&BNDF QUANTITY='CONVECTIVE HEAT FLUX'/

&TAIL /


<img width="918" alt="Screenshot 2023-04-28 at 9 27 54 AM" src="https://user-images.githubusercontent.com/7450069/235160818-f04de2c6-76a7-451b-a5d2-dfded8d78459.png">

**Reply by Weggins** [reply]
Dear sir, I use Windows computer. Now I set the load of vegetation is 0.6 and the fire will extinguish.

**Reply by Weggins** [reply]
&HEAD CHID='canyon'/
&TIME T_END=600/
&DUMP RENDER_FILE='canyon.ge1', COLUMN_DUMP_LIMIT=.TRUE.,DT_RESTART=300.0, DT_SL3D=0.25/



&MESH ID='canyon', IJK=80,70,50, XB=-4.0,4.0,-1.0,6.0,0.0,5.0/



&SURF ID = 'GRASS'
      MATL_ID = 'GENERIC VEGETATION'
      MASS_PER_VOLUME = 0.6
      THICKNESS= 0.5
      MOISTURE_FRACTION= 0.058
      SURFACE_VOLUME_RATIO = 12240
      EMISSIVITY = 0.99
      MINIMUM_BURNOUT_TIME=10

      RGB = 122,117,48
  /

&REAC FUEL='FUEL VAPOR', C=2.10, H=6.20, O=2.16, SOOT_YIELD=0.01, HEAT_OF_COMBUSTION=17425., IDEAL=T /

&SPEC ID='WATER VAPOR' /

&MATL ID                    = 'GENERIC VEGETATION'
      DENSITY               = 500.
      CONDUCTIVITY          = 0.2
       SPECIFIC_HEAT_RAMP    = 'c_v'
      A                     = 1040.
      E                     = 61041.
      NU_SPEC               = 0.75
      SPEC_ID               = 'FUEL VAPOR'
      NU_MATL               = 0.25
      MATL_ID               = 'CHAR'
      HEAT_OF_REACTION      = 418. /

&MATL ID                    = 'CHAR'
      DENSITY               = 300.
      CONDUCTIVITY          = 0.052
       SPECIFIC_HEAT_RAMP    = 'c_v'
      SURFACE_OXIDATION_MODEL = T
      A                  = 465.
      E                  = 68000. 
      SPEC_ID               = 'PRODUCTS','AIR'
      NU_SPEC               = 8.13,-7.17
      MATL_ID               = 'ASH'
      NU_MATL               = 0.04
      HEAT_OF_REACTION      = -25000. /

&MATL ID                    = 'ASH'
      DENSITY               = 67.
      CONDUCTIVITY          = 0.1
       SPECIFIC_HEAT_RAMP    = 'c_v' /

&RAMP ID='c_v', T=  0., F=1.1 /
&RAMP ID='c_v', T=200., F=2.0 /
&RAMP ID='c_v', T=800., F=2.0 /



&GEOM ID='terrain', SURF_ID='GRASS', IS_TERRAIN=T,BNDF_GEOM=T, 
   VERTS=
-3.5,0.0,2.0,
-3.5,6.0,2.0,
-0.1,0.0,0.0,
-0.1,6.0,0.0,
0.1,0.0,0.0,
0.1,6.0,0.0,
3.5,0.0,2.0,
3.5,6.0,2.0,

FACES= 
1,4,2,1,
1,3,4,1,
3,6,4,1,
3,5,6,1,
5,8,6,1,
5,7,8,1,
/
&SURF ID='fire',HRRPUA=1000,RAMP_Q='RAMPIGN',RGB=255,0,0,/
&RAMP ID='RAMPIGN',T= 0,F=0 /
&RAMP ID='RAMPIGN',T= 5,F=0 /
&RAMP ID='RAMPIGN',T= 25,F=1 /
&RAMP ID='RAMPIGN',T= 26,F=0 /
&VENT XB=-0.1,0.1,0.5,0.7,0.0,0.0,SURF_ID='fire', GEOM=T/

&VENT MB='XMIN', SURF_ID='MIRROR' /
&VENT MB='XMAX', SURF_ID='MIRROR' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /
&VENT MB='ZMAX', SURF_ID='OPEN' /


&BNDF QUANTITY='WALL TEMPERATURE'/
&BNDF QUANTITY='WALL THICKNESS'/
&BNDF QUANTITY='BURNING RATE'/
&BNDF QUANTITY='RADIATIVE HEAT FLUX'/
&BNDF QUANTITY='CONVECTIVE HEAT FLUX'/

&TAIL /

**Reply by Weggins** [reply]
The fire can spread with 0.6 kg/m2 in my experiment and the height of the vegetation is about 5 cm.

**Reply by ericvmueller** [reply]
I copied your input above but increased the output of the burner by having full output between 2-25 s:
&RAMP ID='RAMPIGN',T= 0,F=0 /
&RAMP ID='RAMPIGN',T= 5,F=1 /
&RAMP ID='RAMPIGN',T= 25,F=1 /
&RAMP ID='RAMPIGN',T= 26,F=0 /

I get spread but only upslope. You can try different ignition energy or duration and maybe you will get better results. The grid resolution may also be an important factor, especially without wind. The boundary fuel model does not propagate energy forward through the fuel bed, only downward. So it relies on energy transfer from the gas-phase above to propagate.

![image](https://user-images.githubusercontent.com/13366803/235200839-165bb3e0-4a13-42db-8561-d5fe8340aa90.png)

**Reply by Weggins** [reply]
I have taken your suggestion to increase the grid density. The model has taken a few days to run but the result is not good. Moreover, I also find another problem. I want to know why fire can ignite at the edge of the terrain.
![fireline3](https://user-images.githubusercontent.com/131326636/236092789-b8801447-7bd4-49fe-96bd-5c383a36c6cc.png)
![fireline4](https://user-images.githubusercontent.com/131326636/236092809-e004c9ec-0a67-42c4-bdf0-d6135140ec81.png)
![fireline5](https://user-images.githubusercontent.com/131326636/236092811-b876e925-edd8-4f6c-aade-0a8c7addff52.png)

**Reply by ericvmueller** [reply]
Marcos is working on the determining the best default behavior (https://github.com/firemodels/fds/issues/11781). For now, you need to specify `BACKING = 'VOID'` on the `SURF` line to avoid heat transfer through the terrain.

**Comment by Weggins** [comment]
Today I run the model by another slope, and the fire is not symmetric. What can I do to solve this problem?
![微信截图_20230505181444](https://user-images.githubusercontent.com/131326636/236433664-584de6fd-3e1b-4932-9266-4cdc5f40cacb.png)

**Comment by marcosvanella** [comment]
Hi @Weggins, asymmetries can be due to initial conditions, boundary conditions, fire loading, discretization or feedback among these. Post the input file so we can look at this here.
Also, do me a favor, make your ZMIN boundary of type OPEN. Do you see a hole across the bottom of the valley? Seems your calculation is using the external wall SURF_ID instead of the geometry.
________________________________

**Comment by Weggins** [comment]
Thank you, sir
I just add ZMIN to OPEN and send the file. 
[canyon.txt](https://github.com/firemodels/fds/files/11406620/canyon.txt)

**Reply by Weggins** [reply]
There is still a problem that the fire just spreads upslope. I guess the reason is that vegetation cannot receive radiation and convective heating except upslope.

**Comment by marcosvanella** [comment]
@Weggins this is what I see with latest source. I took out the open vent from the zmin boundary. It seems that any asymmetry initially is affecting the fire growth on one side, which affects the flowfield due to buoyancy and feeds back into higher upslope velocity and fire growth in that side. Do you actually have experimental results for these cases?


https://user-images.githubusercontent.com/7450069/236854176-75e628da-6530-4d21-adfd-fd5b6aef9192.mp4



Run the case with latest bundle (https://github.com/firemodels/test_bundles/releases/tag/FDS_TEST), we haven't been able to reproduce what you see in the valley line. Make sure that in smokeview you disable external boundary surfaces (go to show/hide -> geometry -> surfaces -> Exterior).
Another option is make the mesh:
&MESH ID='canyon', IJK=80,60,60, XB=-4.0,4.0,0.0,6.0,-0.05,2.95/ 
and let me know if you see the hole through the valley line.

**Reply by Weggins** [reply]
I can show some photos in experiment. 
![IMG_3489](https://github.com/firemodels/fds/assets/131326636/5ea5f3b0-615e-47eb-ba46-dd616e5e3208)
![IMG_1932](https://github.com/firemodels/fds/assets/131326636/ba54b156-cdce-4ca1-aaab-71bd38583ac7)

**Reply by Weggins** [reply]
There is a new problem as shown in the follwing
![微信图片_20230513004219](https://github.com/firemodels/fds/assets/131326636/2d64e2fa-9453-4d20-8d05-cae5ba96fba1)

**Comment by mcgratta** [comment]
This means that radiation absorption within a solid is only possible for flat surfaces, not spheres or cylinders.

**Comment by marcosvanella** [comment]
It seems the fire line spreads at slower rate towards the flanks in the experiment.  How did you get the material properties you are using for the boundary fuel model? Do you have a case with no slope (flat table) and same fuel bed? Also, how thick is the fuel bed in cm?

The other question is grid sensitivity, if you refine the mesh, do you see much of a change in the results? Split the domain in several meshes using a MULT line and run in parallel, i.e:
&MULT ID='MyMult', DX=2., DY=2., I_UPPER=3, J_UPPER=2 / # 12 meshes, run with 12 MPI processes, see guide.
&MESH ID='canyon', IJK=40,40,60, XB=-4.0,-2.0,0.0,2.0,0.0,3., MULT_ID='MyMult'/ ! Cell size 0.05 m

**Reply by Weggins** [reply]
The material properties are used according the samples. We have a test in a flat to get the fire spread rate (R0), which is about 0.15cm/s. The thick is about 5cm.
I tried to increase the grid sensitivity to 5cm, but the fire still spread upslope. The reason may be that the  boundary fuel model do not consider heat conduction. The fire spread upslope is caused by the heat from gas combustion, while the fuel downslope and flat cannot absorb the heat from gas combustion.
Now I'm trying to use &OBST to estalish a canyon, rather than &GEOM, and the fire can spread in all direcation.
![微信截图_20230511190959](https://github.com/firemodels/fds/assets/131326636/601a85cd-478c-430b-8187-bb021fa5bf72)
![微信图片_20230512203248](https://github.com/firemodels/fds/assets/131326636/55ca277a-e6e1-447f-ae94-eea4a302009c)

**Reply by Weggins** [reply]
Now I want to use Lagrangian Particle to discribe the fuel bed, the fuel used is pine needles. Is there a suitable sample to use for my model?

**Comment by marcosvanella** [comment]
Thank you @Weggins. Please post your obst input file so we can look at differences here.

**Reply by Weggins** [reply]
[canyon1.txt](https://github.com/firemodels/fds/files/11471645/canyon1.txt)
&OBST is also used to describe the vegetation, but I think it is not suitable. The best way is to use Lagrangian Particle model.
